moveit_core/robot_state/src/robot_state.cpp

```diff
@@ -1151,6 +1151,21 @@ bool RobotState::getJacobian(const JointModelGroup* group, const LinkModel* link
   Eigen::Vector3d joint_axis;
   Eigen::Affine3d joint_transform;
 
+  // Travel up the chain to ensure link is a part of group.
+  const LinkModel* given_link = link;
+  while (not group->hasLinkModel(link->getName()))
+  {
+    link = link->getParentLinkModel();
+    if (not link)
```

Comment:
It was already checked in [line 1126](https://github.com/ros-planning/moveit/pull/877/files#diff-15d4dc0e9e1eee143f7576803dfe8ea2L1126) that link is updated by this group.

Reply:
Yes, but it doesn't check that this link/its parent joint are a part of the group, which is an assumption made several lines down at `group->getVariableGroupIndex(pjm->getName());` (see below comment for full explanation.